frontend/tests/controllers/UserRankTest.php

```diff
@@ -806,4 +806,168 @@ public function testUserWithIdentitiesRank() {
             $this->assertSame($entry['ranking'], $i + 1);
         }
     }
+
+    /**
+     * Test that site-admins are excluded from user rankings
+     * even if they solve more problems than regular users.
+     */
+    public function testSiteAdminExcludedFromUserRanking() {
+        // Create a regular user who solves few problems
+        ['identity' => $regularIdentity] = \OmegaUp\Test\Factories\User::createUser();
+
+        // Create a site-admin user who solves many problems
+        ['identity' => $adminIdentity] = \OmegaUp\Test\Factories\User::createAdminUser();
+
+        // Admin solves 10 problems
+        for ($i = 0; $i < 10; $i++) {
+            $problem = \OmegaUp\Test\Factories\Problem::createProblem();
+            $runData = \OmegaUp\Test\Factories\Run::createRunToProblem(
+                $problem,
+                $adminIdentity
+            );
+            \OmegaUp\Test\Factories\Run::gradeRun($runData);
+        }
+
+        // Regular user solves only 2 problems
+        for ($i = 0; $i < 2; $i++) {
+            $problem = \OmegaUp\Test\Factories\Problem::createProblem();
+            $runData = \OmegaUp\Test\Factories\Run::createRunToProblem(
+                $problem,
+                $regularIdentity
+            );
+            \OmegaUp\Test\Factories\Run::gradeRun($runData);
+        }
+
+        // Update ranks
+        \OmegaUp\Test\Utils::runUpdateRanks();
+
+        // Get user rankings
+        $response = \OmegaUp\Controllers\User::getRankByProblemsSolved(
+            null,
+            '',
+            1,
+            100
+        );
+
+        // Verify regular user appears in ranking
+        $regularUserFound = false;
+        $adminUserFound = false;
+
+        foreach ($response['rank'] as $rankedUser) {
+            if ($rankedUser['username'] === $regularIdentity->username) {
+                $regularUserFound = true;
+            }
+            if ($rankedUser['username'] === $adminIdentity->username) {
+                $adminUserFound = true;
+            }
+        }
+
+        $this->assertTrue(
+            $regularUserFound,
+            'Regular user should appear in user ranking'
+        );
+        $this->assertFalse(
+            $adminUserFound,
+            'Site-admin should not appear in user ranking'
+        );
```

Comment:
Here we can avoid the iteration to check whether the usernames appear or not by using `assertContains` and `assertNotContains` directly:
```suggestion
        $rankedUsers = array_column($response['rank'], 'username');
        $this->assertContains(
            $regularIdentity->username,
            $rankedUsers,
            'Regular user should appear in user ranking'
        );
        $this->assertNotContains(
            $adminIdentity->username,
            $rankedUsers,
            'Site-admin should not appear in user ranking'
        );
```

The result should be the same

Reply:
DONE DONE 👌🏻